proxy/envoy/route.go

```diff
@@ -83,6 +83,22 @@ func buildInboundWebsocketRoute(rule *proxyconfig.RouteRule, cluster *Cluster) *
 	return route
 }
 
+func buildInboundRoute(config model.Config, rule *proxyconfig.RouteRule, cluster *Cluster) *HTTPRoute {
+	route := buildHTTPRouteMatch(rule.Match)
```

Comment:
Can you just fold this and the buildInboundWebsocketRoute into one single function
`buildInboundRoute` ? You can then check 
`if rule.WebsocketUpgrade {` and set the route.WebSocketUpgrade to true.. This way, the code is compact.. This should also kill the if-else loop that you have above in buildInboundListener

Reply:
Why do we need to know rules at inbound location? I understand the hack for websockets was placed due to the fact that Envoy cannot multiplex regular HTTP with websockets, but we should not rely on that.

Reply:
Its for tagging the traces with the operation names.. The thing is, multiple routes could lead to the same inbound route like / . In this case, the operation name becomes pointless right? @objectiser 
You might as well have "inbound" as the operation name.

Reply:
Not sure if pointless, but less useful - although using the current approach it still gives them the ability to define the span/operation name - even if all endpoints for the same destination use the same name (initially). But from there they then have the ability to refine the routes and provide more specific names.